tests/unit/db/vertica/test_unloader_no_aws_creds.py

```diff
@@ -49,7 +49,7 @@ def test_unload_with_no_aws_creds(self,
 
         mock_column = Mock(name='column')
         mock_db.dialect.get_columns.return_value = [mock_column]
-        with self.assertRaises(NotImplementedError):
+        with self.assertRaises(AttributeError):
```

Comment:
where is this behavior changing and why? is the attribute error an internal thing from sqlalchemy getting surfaced?

Reply:
Fixed this, now using a MagicMock to reproduce NotImplementedError